### DISCUSSION TITLE
NextJs api calls always result in 405

### DISCUSSION THREAD

**Discussion by pandydavey** [vercel/next.js#21068] [Help]
I'm using Next.js and locally i dont have any issues. But when i deploy to either my preview or my prod builds i cant seem to call any of my api routes. Or rather, i can call them, but i always get 405 errors. I cant seem to figure out what the issue is. I've read the documentation but cant seem to see anything relating to this problem. 

Can someone help me try and debug this situation, or if they've seen the problem and fixed it can you let me know what it is please. 

```
import { NextApiResponse } from 'next';
/**
 * Get the User details
 */
export default async function handler(
  res: NextApiResponse,
): Promise<any> {
	res.statusCode = 200;
	res.setHeader('Content-Type', 'application/json');
	res.end(JSON.stringify({ name: 'John Doe' }));
};
```

this is my super simple API endpoint. 

this is how i'm calling it 

`const res = await fetch(url, options);`

where options is: 

```
{
	method: "POST",
	body: {"email":"example@example.com","name":"","rss":"","csrfToken":""}
}
```

as i've said locally i dont have any issues at all.

**Comment by pandydavey** [comment]
ok, so, i think i know the issue, it was always going to be an issues on my end lets be honest. But this is here incase anyone else finds themselves in the same pickle. 

when i first started nextjs (6 months ago) you needed to add when client side fetching `${process.env.NEXT_PUBLIC_SITE}/api/route` on closer inspection this was adding the url twice and thus causing an issue. removing this public site env value has fixed my problem. Sorry and thanks for anyone who read this.

**Reply by m3c-ode** [reply]
So what di you put instead?
